<launch>
  <arg name="sim" default="false" />
  <arg name="isolated_arm_driver" default="true" />
  <arg name="scene" default="true" />
  <arg name="octomap" default="true" />
  <arg name="grasp_controller" default="true" />
  <arg name="initial_z" default="0" />
  <arg name="start_home" default="$(arg sim)" />
  <arg name="scene_file" default="$(find mujoco_models)/mujoco.scene" />
  <arg name="camera" default="ensenso" />
  <arg name="arm_group" default="right_arm_and_manipulator" />
  <arg name="description_repo_path" default="$(find sr_utl5)" />
  <arg name="load_grasps" default="true" />
  <arg name="manipulator_file" default="$(find sr_utl5)/config/manipulator_hand.yaml" />
  <arg name="arm_positions_file" default="$(find sr_utl5)/config/arm_positions.yaml" />
  <arg name="cylinder" default="false" />
  <arg name="box" default="false" />
  <arg name="robot_model_path" default="$(find mujoco_models)/models/tote_env_test.xml" unless="$(arg cylinder)" />
  <arg name="robot_model_path" default="$(find mujoco_models)/models/ur10_fh_environment_cylinder.xml" unless="$(arg box)" />

  <!-- Robot stuff -->
  <include file="$(find fh_robot_launch)/launch/fh_ur10_and_fh2_mujoco.launch">
    <env name="MESH_WORKSPACE_PATH" value="$(find sr_utl5)/models/" />
    <arg name="sim" value="$(arg sim)"/>
    <arg name="scene" value="$(arg scene)" />
    <arg name="scene_file" value="$(arg scene_file)" />
    <arg name="initial_z" default="$(arg initial_z)" />
    <arg name="grasp_controller" value="$(arg grasp_controller)" />
    <arg name="robot_description" value="$(find fh_2_ur10_moveit_config)/config/fh_v2_ur10_with_hand.urdf.xacro" />
    <arg name="robot_model_path" value="$(arg robot_model_path)"/>
  </include>

  <!-- Recognized object mocap -->
  <node pkg="sr_vision_mocks" type="mock_recognized_objects.py" name="mock_recognized_objects" output="screen"/>

  <!-- Grasping stuff -->
  <param name="arm_group" value="$(arg arm_group)" />
  <param name="grasp_action_name" value="H0_grasp_controller/grasp_action" />
  <param name="description_repo_path" value="$(arg description_repo_path)" />
  <rosparam command="load" file="$(arg manipulator_file)"/>
  <rosparam command="load" file="$(arg arm_positions_file)"/>

  <!-- Needed for the pipeline to work -->
  <include file="$(find sr_manipulation_grasp_selector)/launch/grasp_selector_with_db.launch" />

  <node if="$(arg box)" pkg="sr_utilities_common" type="timed_roslaunch.sh" args="20 sr_manipulation_db fill_grasp_db.launch grasps_file_path:='$(find sr_manipulation_state_machine)/config/grasps.yaml'" name="timed_load_grasps" output="screen"/>
  <node if="$(arg cylinder)" pkg="sr_utilities_common" type="timed_roslaunch.sh" args="20 sr_manipulation_db fill_grasp_db.launch grasps_file_path:='$(find sr_manipulation_state_machine)/config/grasps_cyl_power_mujoco.yaml'" name="timed_load_grasps" output="screen"/>
</launch>
